circuits/cpp/src/aztec3/circuits/kernel/public/native_public_kernel_circuit_no_previous_kernel.cpp

```diff
@@ -13,15 +13,27 @@
 
 namespace {
 
+using aztec3::utils::is_array_empty;
+using CircuitErrorCode = aztec3::utils::CircuitErrorCode;
+
 /**
  * @brief Initialises the circuit output end state from provided inputs
+ * @param composer The circuit composer
  * @param public_kernel_inputs The inputs to this iteration of the kernel circuit
  * @param circuit_outputs The circuit outputs to be initialised
  */
-void initialise_end_values(PublicKernelInputsNoPreviousKernel<NT> const& public_kernel_inputs,
+void initialise_end_values(DummyComposer& composer,
+                           PublicKernelInputsNoPreviousKernel<NT> const& public_kernel_inputs,
                            KernelCircuitPublicInputs<NT>& circuit_outputs)
 {
     circuit_outputs.constants.tx_context = public_kernel_inputs.signed_tx_request.tx_request.tx_context;
+
+    // If the nullifiers array is not empty a change was made and we need to rework this
+    composer.do_assert(is_array_empty(circuit_outputs.end.new_nullifiers),
```

Comment:
Same as the private circuit.

Reply:
Makes sense. I just tried implementing this and there is a bit of an issue here because `public_kernel_inputs` in `update_public_end_values` contains `signed_tx_request`. Would the correct approach be to try casting it or would we want to create a custom `update_public_end_values` function just for the kernel version with no previous input?

I am a C++ noob so I could use some feedback here.